server/src/main/java/org/opensearch/index/translog/RemoteFsTranslog.java

```diff
@@ -58,6 +59,8 @@ public class RemoteFsTranslog extends Translog {
 
     // clean up translog folder uploaded by previous primaries once
     private final SetOnce<Boolean> olderPrimaryCleaned = new SetOnce<>();
+    //
+    private final ReentrantLock remoteGenerationDeletionLock = new ReentrantLock();
```

Comment:
Should we use a `RateLimiter` instead. Also we can simplify deletion with just an AtomicBoolean flag right?

Reply:
Not using Reentrant lock anymore. Have made multiple changes. Please review again and let me know your feedback.